src/util/irep.h

```diff
@@ -187,13 +190,13 @@ class irept
   #endif
 
   const irep_idt &id() const
-  { return read().data; }
```

Comment:
No, please don't. This "id" is the only way to actually store data in an irep (other than the mere nesting structure).

Reply:
Ah, is that the reasoning? That's probably worth a comment.

I think the thing that annoys me is having two fields called `data`, so another approach would be to rename the `data` field in `irep`. Maybe `dt_ptr`, since it is a pointer to a `dt`. What do you think of that?

Reply:
> Ah, is that the reasoning? That's probably worth a comment.

It's my guess that it is - only @kroening will know for sure :)  Would you mind adding a comment to that effect?

> [...] so another approach would be to rename the data field in irep [...]

Ah, understood. In line with an earlier comment by @thk123: don't abbreviate variable names, so maybe use `data_ptr` (oh, well, that's still an abbreviation, is it?).

Reply:
I have added that comment.

@martin-cs pointed out to me that when sharing is turned off `data` won't be a pointer. I also see that @reuk has a pending PR which refactors irep, so maybe now isn't the best time to rename the field. I'll mention my thoughts to him and see what happens.

Reply:
How about you add your commits onto his branch (by creating a pull request against his fork)? Then we can avoid rebasing hell...

Reply:
I've done that